### PR TITLE
HPCC-15763 Increase default physical parallel delete limit.

### PR DESCRIPTION
**Pull request by jakesmith**
Make default the number of parts.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@AttilaVamos - please review

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -3308,7 +3309,7 @@ class CDistributedFile: public CDistributedFileBase<IDistributedFile>
                 throw exceptions.getClear();
         }
     }
-    bool removePhysicalPartFiles(IFileDescriptor *fileDesc, IMultiException *mexcept)
+    bool removePhysicalPartFiles(IFileDescriptor *fileDesc, IMultiException *mexcept, unsigned parallelPhysicalDeletes=NotFound)
```
Comment (AttilaVamos) [inline dali/base/dadfs.cpp]: I think something like 'numberOfParallelPhysicalDeletes' or 'numberOfParallelPhysicalDeleteThreads' would be better than 'parallelPhysicalDeletes'
I think the 'NotFound' is not the best self documented default value for 'parallelPhysicalDeletes'. Something like 'UnSet' or 'NotSet' would be better (or simple 0 if the parameter renamed to 'numberOfParallel...')

**Comment by AttilaVamos** [comment]
@jakesmith I have only one, self documentation related comment.

**Comment by jakesmith** [comment]
@AttilaVamos - updated.

**Comment by AttilaVamos** [comment]
@jakesmith It seems good.

**Comment by jakesmith** [comment]
squashed.

@richardkchapman - ready to merge.

**Comment by richardkchapman** [comment]
Why was it limited before?

Would the number of physical nodes be better than the number of parts?

**Comment by jakesmith** [comment]
Just very old / low limit I think.
# of nodes vs # parts - might be sensible. It depends why on some file systems (e.g. ext3) it is so slow to delete large files and if e.g. it is approx twice as fast to delete 2 large files on the same FS or not.
My guess is, you do geneally gain from parallel deletes on a FS.

**Pull request merged**